### DISCUSSION TITLE
Nemo - Merge "title bar" with "tool bar"

### DISCUSSION THREAD

**Discussion by Jopp-gh** [linuxmint/discussions#425] [Ideas]
Luckily I've a big screen, but still, the head part of Nemo's windows is too big and the benefits are none. Let me explain why:

- the name of active directories is already shown on tabs or in the path bar. 
- minimize, zoom and close don't use much space and can easily be merged with the tool bar
- you can grab windows with a shortcut key, example: super+left mouse click, hold and drag

I see no reason to have a header double the size, when it would work perfectly only half the size. Also, many modern Linux apps merged already name and tool -bar: Pix, calendar, Planify, lutris...

**Comment by Hilyxx** [comment]


**Reply by Secret-chest** [reply]
Nemo didn't exist in 2007

**Reply by Hilyxx** [reply]
"Nemo didn't exist in 2007"  Back to the future... Typing error.

**Comment by Secret-chest** [comment]
What about the menus? We can't lump them all in one, there are too many

**Reply by Jopp-gh** [reply]
Right, there are many, I forgot to include them because they're useful on occasion only. In my case they are hidden and open with the shortcut `alt`. 

Add a "hamburger-menu" (pop-up menu) on the left or right and add there all commands. Since submenus expand on hover, navigation through a "hamburger-menu" isn't a problem.

**Reply by Jopp-gh** [reply]
Fine for me too. 

I suggested a hamburger popup menu because in my opinion, it's a very quick way to access all kinds of submenus/commands

**Reply by cplittlfield** [reply]
Having the directory name in the title is a very handy feature for those of us who have multiple Nemo windows open at once.  Makes it easy to pick the desired window out of several that are open.

One of the refreshing things about Linux Mint is its conservative look.  I can no longer use Pix, for example, because they did away with the menu, and it is now impossible to find what commands are available when I need to do something.  I don't have the time to spend ten minutes looking for a feature that used to be easily findable eighteen months ago.

The "old-fashioned" window layout was designed by smart people for good reasons, and the reason it has endured for fifty years is that it works.  I understand people want a cool new look, but I personally would like Linux Mint to remain a refuge for people who find "cool" a hindrance to efficient work flow.  (Don't even get me started on skinny, disappearing scroll bars, lol!)

**Comment by SebastJava** [comment]
I get your idea and I approximately agree with it. But I would much prefer this:
Nemo - Merge **title bar** with **menu bar**

![Nemo-headerbar](https://github.com/linuxmint/discussions/assets/31940713/6fd1503b-00cd-4151-8c13-8d9077f655d6)

**Reply by SebastJava** [reply]
Materia-compact theme here... :)

**Reply by SebastJava** [reply]
@Jopp-gh  I suggest you change your title on your OP: Nemo - Merge **title bar** with **menu bar**

**Reply by Jopp-gh** [reply]
My post wants to discuss the merge between "title bar" and "tool bar".

I didn't talk about the "menu bar" in particular, because: you can hide and show that bar. Just use "alt" on your keyboard. 

If you really need to see that bar all times, then I suggest to

- place the "menu bar" between the tool bar on top and the main browsing area below (where you see folders and files).

- place the menu bar in the left sidebar (where we have bookmarks, user folders of home, etc.) - that's a bit tight, but a test wouldn't hurt and show immediately if comfortable, easy to access for daily usage.

- finally you could place the menu bar as hover-popup in the main browsing area itself that's be yet another (possible) way. Just talking about possibility, not "like".

From a usage view, I myself always hide the menu bar and use shortcuts instead. 
Shortcuts are fast.

Also, a file browser is an essential interface on any computer and because users are constantly exposed to file navigation, users *should* have a good motivation to memorize some of their most important / frequently used, file browsing shortcuts - because shortcuts are faster than digging all times into menus - I would say, shortcuts fall under the keyword "productivity". 

Again, in my view I would prefer a simple "hamburger menu" button - this does integrate very well with a tool bar too.
If - just if - you put there all menu bar items, you can access all your commands in one place, with  popup menus that open on hover. Conceptually, it's the same as with a right mouse click to open a context menu. Easy and fast.

No digging, clicking around like in old times.  Time is valuable.

**Reply by Secret-chest** [reply]
You can learn shortcuts, and I do, but don't force everyone. There are too many items to not use a bar.

**Reply by Jopp-gh** [reply]
Please don't interpret things into my post I didn't write - I don't "force" anybody. I just try to communicate. 

In fact, if you care to read my whole last answer, I did **include more possibilities** and hope to discuss this matter in a **reasonable and emotional-free way**. 
Of course I also describe my own views and won't excuse myself for being myself.

**Comment by Qronikarz** [comment]
I made a visualization, does it resemble what you wanted?
Here's how it looks now:
![nemocurrent](https://github.com/linuxmint/discussions/assets/84787215/7c55d407-c013-438a-b3de-718b264dd55c)

Here's how it would look:
![nemocompact](https://github.com/linuxmint/discussions/assets/84787215/23151033-6605-48cb-a763-630e7c47cbfa)

As was said before, the menu can still be shown using alt button. Just like in Firefox for example. Another thing that could be compacted without taking functionality away are the grid, list and column view buttons. They can just be merged into one button with dropdown next to it and this way there's more place for the folder path.

**Reply by Jopp-gh** [reply]
Your mock-up looks great! 

So far I didn't think about the grid, list and column view buttons, interesting.
I should start to add mockups too. 👍

**Reply by Jopp-gh** [reply]
Here's my mock-up, the space gain is substantial.
![Nemo-mockup-1_122217](https://github.com/linuxmint/discussions/assets/48620536/bfceeb4e-6ade-4aeb-a613-1280262d69f8)

Furthermore, I added a tag panel for another browsing experience (tag panel button circled in the "status bar") discussed in other posts.

**Reply by Secret-chest** [reply]
If we do have this, do we need the folder path in the middle? It is already there.

**Reply by Qronikarz** [reply]
Actually we don't need it. I just put it there to show that there's still plenty of space and to make it certain which folder are you in. However the idea below of adding accent color to the breadcrumbs is way better.
https://github.com/orgs/linuxmint/discussions/425#discussioncomment-9545665

**Comment by DirkHaar** [comment]
> Luckily I've a big screen, but still, the head part of Nemo's windows is too big and the benefits are none. Let me explain why:
> 
>     * the name of active directories is already shown on tabs or in the path bar.
> 
>     * minimize, zoom and close don't use much space and can easily be merged with the tool bar
> 
>     * you can grab windows with a shortcut key, example: super+left mouse click, hold and drag
> 
> 
> I see no reason to have a header double the size, when it would work perfectly only half the size. Also, many modern Linux apps merged already name and tool -bar: Pix, calendar, Planify, lutris...

I f***ing hate programs with a title bar. With 20 apps open when working which sometimes have several unpinned windows or dialogs open it is a big PITA to figure out which belongs where, esp. on dual or multi monitor setups.
The title in the middle is recognized in 1/10 of a second even without focusing. 

Edit:
I meant: " I f***ing hate programs **WITHOUT** a title bar". Hopefully the rest of the text makes that clear.

**Reply by Jopp-gh** [reply]
You got a point, but scale@cinnamon.org and expo@cinnamon.org have already this feature implemented.

If you aren't in overview mode (using the applets above) then Nemo would need labels.

Edit.
Removed irrelevant details.

**Reply by Jopp-gh** [reply]
Here an idea:
![Nemo-mockup-2_122735](https://github.com/linuxmint/discussions/assets/48620536/b680a8dd-bfa8-46ea-9c34-9674f8f0b215)

Added a floating label on top of the window. Maybe an idea.

**Reply by Secret-chest** [reply]
It takes up the same amount of space but is less practical.

**Reply by Jopp-gh** [reply]
The idea is to make it "floating" above, with some offset from the tool bar, without a "collision box" or how this stuff is called.

**Reply by Secret-chest** [reply]
No. I don't care how nice it looks, but it would work weirdly. I would accept putting it in the menu bar without an offset but this is too much. It even takes up more space than it does currently if you hide the menubar.

**Reply by Jopp-gh** [reply]
Here another idea.
Just add a highlight (accent color) to the name of the current directory to identify quickly the window you're in. (See the path bar highlighted with Mint's accent colour)

![Nemo-mockup3_122100](https://github.com/linuxmint/discussions/assets/48620536/194c90b1-a8a3-47f5-86b9-cbb56e76170b)

**Reply by Secret-chest** [reply]
If the full accent is too much we can underline it with the accent.

I don't like where the menu bar is, it should be full-width.

**Reply by Jopp-gh** [reply]
Yes, underline could be enough.

>In my own software projects I always allow new features to run for some days / a week, to see how these changes impact usage in good or in bad for a better understanding. 

**About the menu bar**. 
If Nemo would introduce a different menu bar, then probably people would ask for the same placement in all  apps too. 
I myself would love a popup button in the tool bar instead of the menu bar. 

>Example, **Gimp** allows to access the menu bar from the context menu (right click on empty space) in full screen mode. I know right click isn't "a button" on the ui, still it would work very similarly.

**Comment by moloch1994** [comment]
I disagree. I've used Gnome in the past, and often there isn't much space to grab the window by when you wanna drag it. It requires more precision, making you less efficient.

Why not just use Nautilus if you prefer headerbars?

**Reply by marco4mp** [reply]
That is not true, the window can be dragged from any point on the top bar (CSD)

**Reply by Jopp-gh** [reply]
I see. Yes, to grab and move windows is yet another feature to consider.

Months ago, when I first started to use Mint, I didn't know that Nemo allows us to define a shortcut to grab-move any app window on any surface of a window.

Today I simply click and hold `super+left mouse button` to grab a window. Then I use my mouse to move my window. A really convenient feature imo, this way you aren't confined anymore to the title bar. 

Agreed, the toolbar is a bit tight to drag.
Maybe a new toolbar "pan" button could do the job. With a pan button enabled and temporarily suspending all other buttons, you could drag Nemo's window on any surface too.

**Reply by Jopp-gh** [reply]
Here a very quick mock-up.
See the circled (hand) button
Ideally, click and hold the pan button then move your window. Basically, almost the same as grabbing the title bar.

![Nemo-mockup4_113130](https://github.com/linuxmint/discussions/assets/48620536/3dcc2147-b701-4e22-a0e2-c0d3a134f48f)

**Reply by moloch1994** [reply]
yeah, I'm aware of that feature. It's by holding alt by default. But a lot of the times it's either inconvenient for me to use the keyboard or impossible (e.g. when I have my computer connected to the TV). I still don't see why not keep Nemo the same and just use Nautilus if you want headerbars? I get that it lacks the "open as root" feature, but it can be installed quickly. The search is a bit less convenient, but other than that I think Nautilus works more or less the same.

Personally I like that Cinnamon stays close to older Windows designs and doesn't follow the complete Mac'ification that is happening in Windows 11 and many Linux desktops. I don't like Mac, and if I wanted something close to it, I'd just use Gnome.

Then again, if you make it optional (like the 'no title bar' feature in Firefox), I wouldn't mind.

**Reply by marco4mp** [reply]
@Jopp-gh @moloch1994 Try the Pix app (default Linux Mint app). Space is not a problem, you can drag the window from anywhere on the top bar.

For example, Newsflash, another app that also uses CSD. The little space available on the top bar is not a problem. Just hold a click and drag.


https://github.com/linuxmint/discussions/assets/52709674/10ed7f33-3116-43b9-914b-c2a704873784

**Reply by Jopp-gh** [reply]
You're right. 
I only suggested a button to grab (Nemo) windows to make it more obvious and easy to move windows. Such a button would help newcomers but maybe, be comfortable too. As symbol I used a hand, bc the usual icon for panning, but it mustn't be a hand. 
 
Sometimes the space in the tool bar can be pretty tight so a button might be a "better" (usability) choice.

**Reply by moloch1994** [reply]
> @Jopp-gh @moloch1994 Try the Pix app (default Linux Mint app). Space is not a problem, you can drag the window from anywhere on the top bar.
> 
> For example, Newsflash, another app that also uses CSD. The little space available on the top bar is not a problem. Just hold a click and drag.
> App.CSD.drag.area.mp4

okay, my bad, you're right, it can be dragged from anywhere

**Comment by leigh123linux** [comment]
-1 to making nemo look as hideous as nautilus.
If you want CSD use nautilus instead!

**Reply by marco4mp** [reply]
Why? What are the cons?

**Reply by Secret-chest** [reply]
> Why? What are the cons?

It removes quick access to all options. How do I access "Connect to Server", for example, without a menu bar?

**Reply by JosephMcc** [reply]
You present it another way. In Nautilus there is an additional bookmark in the sidebar called "Other Locations". You can easily connect to a server from there. That being said, Nemo's menu setup is far more complex but quite a lot things good be presented different or only in the context where it makes sense. I would never be down for having both a header bar and a menu bar though. That's essentially trying to smash together to different UI styles and that's pretty much how it looks.

**Reply by Jopp-gh** [reply]
"Hideous" means, that the top bar looks too compact? 
When I first saw this design on Macs, it felt unusual and not really appealing to me. In hindsight, this compact design is very useful because all features are placed in one bar. And Pop-up buttons can store additional commands, if needed.

I didn't think at "looks" when I started this topic, only at ux experience. There's a lot space wasted on the top of every Nemo window.

**Reply by Secret-chest** [reply]
Macs still have a menu bar though.

**Comment by cplittlfield** [comment]
I'm all for offering more customizability, but please do not _remove_ features that people need.  If you want an option to hide the title bar, I'd support that, as long as there is still a way to display it, because I need it.  We can already hide or display the menu bar as we choose, so that's good.  Even the toolbar and the status bar can be hidden at will.  So all good.

What I have a problem with is making things look sleek and cool, _at the expense of workflow._  For example, skinny, disappearing scrollbars are difficult for my elderly, shaking hands to use, so I **_really_** appreciate that Cinnamon gives me the ability to disable overlay scrollbars and to make scrollbars wider.  I don't mind if people prefer the cooler, sleeker look--I would probably prefer it, too, if I were fifty years younger--so long as there are options to give people like me what we need, as well.

**Comment by Jopp-gh** [comment]
Sometimes threads grow so much, that the original proposal is lost on the way. Just to recap if somebody missed my mock-up:
https://github.com/orgs/linuxmint/discussions/425#discussioncomment-9532944

I fully agree, user interface designs needs to help users to get into a "workflow", a status where everything feels easy and quick to do.

Ux design is therefore in service of users and a "cool look" is nice too, but definitively secondary.

Some people are quicker to control computer interfaces through icons, others prefer readable commands, probably because those people are very fast to read text.

Said this, a compact design looks more minimalistic and even inadequate and it seems to lose its very functionality. 

My proposal was to remove the title bar but not its features, because:
- The path bar shows the name of your current directory, same as the name in the title bar. Nothing is lost.
- And the collapse, maximize  and close buttons are easy to place in the toolbar as well, these icons don't need much space.
- since the menu bar is hidden all times on my setup, I didn't include it.

The reasoning of making interfaces more compact is, to "reduce clutter" so it's easier to enter in a "state of flow". Long time ago I took a look at books of Mihaly Csikszentmihalyi, but I cant remember much of it. 

In my own experience, I like to remove clutter as much as possible to work just with features I really need (keeping extra features hidden). This doesn't mean that I keep always the same habits, over time we change or develop habits too.

**Comment by yixophoenix** [comment]
This still doesn't exist. Can we please get the merged bar option. Atm this is so much wasted space.

**Comment by Secret-chest** [comment]
It would severely limit the address bar, which is very important to me and I want to have it always visible.
